### ISSUE TITLE
Regression: encoding error when parsing a ISO-8859-1 pom with Maven Resolver

### ISSUE THREAD

**Issue by tmortagne** [codehaus-plexus/plexus-utils#163]
Maven Resolver (at least the 3.6.0 one) model processor uses `ReaderFactory#newXmlReader` to create a `XmlStreamReader` and then pass it to `MXParser`.

Problem is that since https://github.com/codehaus-plexus/plexus-utils/commit/1e18ddcc98f318b36449102f3267fd4631cc668b MXParser assumes the input encoding is UTF8 and only changes that if the input reader is a `InputStreamReader` (which is not the case here since it's a XmlStreamReader). So we end up with exception:

```
UTF-8 BOM plus xml decl of ISO-8859-1 is incompatible
```

This whole check is not really needed here since XmlStreamReader is taking care of potential encoding issues already from what I understand.

**Comment by michael-o** [comment]
Will take a look. Is your POM public?

**Comment by tmortagne** [comment]
Yes, in my case the problem was with https://repo1.maven.org/maven2/org/apache/commons/commons-parent/39/commons-parent-39.pom (when resolving the parents of commons-collections:commons-collections:3.2.2).

**Comment by michael-o** [comment]
@belingueres Can you have a look at this?

**Comment by tmortagne** [comment]
I feel like `MXParser#fileEncoding` should be null by default instead of the arbitrary "UTF8" and the check should be done only if it's not null.

**Comment by michael-o** [comment]
> 
> 
> I feel like `MXParser#fileEncoding` should be null by default instead of the arbitrary "UTF8" and the check should be done only if it's not null.

So it should autodiscover the encoding itself?

**Comment by tmortagne** [comment]
> So it should autodiscover the encoding itself?

No, just not check the encoding when it does not know it, same as before this change.

**Comment by michael-o** [comment]
> 
> 
> > So it should autodiscover the encoding itself?
> 
> No, just not compare the encoding with the declaration when it does not know what encoding the reader uses, same as before this change.

Makes sense.

**Comment by belingueres** [comment]
I'll take a look at it.
